### PR TITLE
Start database for testing

### PR DESCRIPTION
**Pull request by jniemii**
When running backend tests, start a dev database. Reset database and migrate before each test.

### CODE REVIEW

File: backend/src/utils/tests.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const timeout = 60;
+
+jest.setTimeout(timeout * 1000);
+
+beforeAll(async () => {
+    await waitForDatabase(pool, timeout);
+});
+
+beforeEach(async () => {
+    await pool.query('BEGIN');
```
Comment (jaakkoha-bytecraft) [inline backend/src/utils/tests.ts]: beforeEach `BEGIN` will fail if any further `BEGIN` calls are made. A more stable way is to run a db reset + migrations on beforeEach.

Comment (jniemii) [reply backend/src/utils/tests.ts]: Resolved by https://github.com/bytecraftoy/aalto-2022/pull/90/commits/0255fed71dedeffdb110237c80c38408fbc84df9

File: backend/src/db/pool.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import { Pool } from 'pg';
+import dotenv from 'dotenv';
+
+if (process.env.NODE_ENV === 'test') {
+    dotenv.config();
```
Comment (itulkki) [inline backend/src/db/pool.ts:7]: Why is this required or when does it happen?
Maybe add some comments to explain this.

Comment (itulkki) [reply backend/src/db/pool.ts:7]: I think I get it now after reading rest of the changes but it would still be good to give a short explanation in the comments

**Review by itulkki — APPROVED** [review]

**Review by hmalmsten — APPROVED** [review]
